app/src/main/kotlin/com/saltedge/authenticator/features/settings/list/SettingsListViewModel.kt

```diff
@@ -84,6 +89,17 @@ class SettingsListViewModel(
                 itemIsClickable = true
             )
         )
+        val darkModeItem = SettingsItemViewModel(
+            iconId = R.drawable.ic_settings_dark_mode,
+            titleId = R.string.settings_system_dark_mode,
+            switchIsChecked = preferenceRepository.darkModeEnabled
+        )
+        return if (isSystemDarkMode()) listItems.apply { add(3, darkModeItem) } else listItems
```

Comment:
don't use position as hardcoded value. calculate value or add the item dynamically.

Reply:
System dark mode selector should switch to disabled state if was clicked `custom dark mode switch` in MainActivity